src/SFA.DAS.EmployerIncentives.UnitTests/Services/EmployerIncentivesPassThroughServiceTests.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using AutoFixture;
+using FluentAssertions;
+using Moq;
+using Moq.Protected;
+using NUnit.Framework;
+using SFA.DAS.EmployerIncentives.Infrastructure.Api;
+using SFA.DAS.EmployerIncentives.Models.PassThrough;
+using SFA.DAS.EmployerIncentives.Services;
+
+namespace SFA.DAS.EmployerIncentives.UnitTests.Services
+{
+    [TestFixture]
+    [Parallelizable(ParallelScope.All)]
+    public class EmployerIncentivesPassThroughServiceTests
+    {
+        [Test]
+        public async Task When_Adding_LegalEntity_Then_should_call_PostAsync_with_correct_url()
+        {
+            var f = new TestsFixture().WithJsonResponseFromInnerApi();
+            await f.Sut.AddLegalEntity(f.AccountId, f.LegalEntityRequest);
+
+            f.VerifyMethodAndPath(HttpMethod.Post, $"accounts/{f.AccountId}/legalentities");
+        }
+
+        [Test]
+        public async Task When_Adding_LegalEntity_Then_should_return_ApiResponse()
+        {
+            var f = new TestsFixture().WithJsonResponseFromInnerApi();
+            var result = await f.Sut.AddLegalEntity(f.AccountId, f.LegalEntityRequest);
+
+            f.VerifyApiResponseIsReturned(result);
+        }
+
+        [Test]
+        public async Task When_Removing_LegalEntity_Then_should_call_DeleteAsync_with_correct_url()
+        {
+            var f = new TestsFixture().WithNoJsonResponseFromInnerApi();
+            await f.Sut.RemoveLegalEntity(f.AccountId, f.AccountLegalEntityId);
+
+            f.VerifyMethodAndPath(HttpMethod.Delete, $"accounts/{f.AccountId}/legalentities/{f.AccountLegalEntityId}");
+        }
+
+        [Test]
+        public async Task When_Removing_LegalEntity_Then_should_return_ApiResponse()
+        {
+            var f = new TestsFixture().WithNoJsonResponseFromInnerApi();
+            var result = await f.Sut.RemoveLegalEntity(f.AccountId, f.AccountLegalEntityId);
+
+            result.Should().NotBeNull();
+            result.StatusCode.Should().Be(f.HttpResponseMessage.StatusCode);
+            result.Json.Should().BeNull();
+        }
+
+        private class TestsFixture
```

Comment:
Probably worth having a discussion with Dan and other EI devs to see if we use this approach?  As far as I'm aware, we aren't using it in any of the other EI solutions?

Reply:
Will do